src/netinet6/route_v6.c

```diff
@@ -2912,6 +2912,10 @@ int handle_route6(vrfid_t vrf_id, uint16_t type, const struct rtmsg *rtm,
 		if (!(nl_flags & NL_FLAG_ANY_ADDR))
 			flags |= RTF_GATEWAY;
 
+    if (IN6_IS_ADDR_V4MAPPED(&gw)) {
```

Comment:
Doing the check only here won't the case of having an ECMP route, such as in the case of having an eiBGP 6PE/6VPE route. In addition, it possibly is confusing to the reader to look gw here, when it may or may not be used. Thus I would suggest pushing this check inside mpath conditional else block just below, and adding a similar check to the appropriate place in the logic called from ecmp6_create.

Reply:
Hi Robert, 

Thanks for your review. 
We understood your corrections suggested. 
In ecmp6_create() we see  nexthop6_fill_mpls() is already having the similar check, so we are adding the check in nexthop6_fill(). 

Thanks,